bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -499,7 +499,10 @@ private State convertState(Widget w, Item i) {
         if (w instanceof Slider || (w instanceof Switch && i instanceof RollershutterItem)) {
             returnState = i.getStateAs(PercentType.class);
         } else if (w instanceof Switch) {
-            returnState = i.getStateAs(OnOffType.class);
+            Switch sw = (Switch) w;
+            if (sw.getMappings().size() == 0) { // only convert plain switches with no mappings
```

Comment:
Such a comment is completely useless: It is dead and only states the obvious! 

Next time you feel the urge to write such a comment: Pleeeeease write a regression test case instead!!!